ambari-web/app/assets/index.html

```diff
@@ -29,10 +29,19 @@
   <script src="javascripts/app.js"></script>
   <script>
       $(document).ready(function() {
-          require('initialize');
-          // make favicon work in firefox
-          $('link[type*=icon]').detach().appendTo('head');
-          $('#loading').remove();
+        require('initialize');
+        // make favicon work in firefox
+        $('link[type*=icon]').detach().appendTo('head');
+        $('#loading').remove();
+
+        //enable initial tooltips
+        $('[data-toggle="tooltip"]').tooltip();
+        //enables tooltips added later
+        const body = document.getElementsByTagName('body')[0];
+        const observer = new MutationObserver(() => {
```

Comment:
It's bad for performance to trigger tooltip on any DOM change, also it may add already removed tooltips(which should be hidden). Use MutationObserver locally with a particular element that needs tooltip.

Reply:
Where exactly would you move this code? I already tried putting it in some API hook of the view and it would not work.

Reply:
You can put tooltip initialization in didInsertElement of an appropriate view.

Reply:
I tried that, but it did not work. I need something that is guaranteed to run after all dynamic items are added to the DOM. didInsertElement() only runs after the template is initially added. If you have, say, table rows that contain tooltips that get added after the initial render, they will not get activated. At least, that was my observation.

Reply:
I meant to add MutationObserver in didInsertElement, and it will observe change only inside table or any other parent element.

Reply:
OK, that worked. Thanks.